### PR TITLE
Hardening/static mhd lib

### PR DESCRIPTION
**Pull request by kzangeli**
MHD - libmicrohttpd is now built from sources and included as a static lib inside the broker.
Fixes issue #1675.

DO NOT MERGE just yet,
this should be tested in more machines and the dockerfile will need adjustment.

### CODE REVIEW

**Comment by fgalan** [comment]
A quick checklist of things that need to be addressed before this PR could be merged:

* Adjust Dockerfile
* Adjust CI jenkins
* Adjust .spec
* Adjust @dmoranj installation scripts

File: doc/manuals/admin/build_source.md [hunk 1]
```diff
@@ -42,6 +42,16 @@ commands that require root privilege):
         tar xfvz v1.0.2.tar.gz
         sudo mv rapidjson-1.0.2/include/rapidjson/ /usr/local/include
 
+* Install microhttpd from sources:
```
Comment (fgalan) [inline doc/manuals/admin/build_source.md]: Maybe we can add some comment about the configure options, e.g:

"Install microhttpd from sources (the `./configure` command below shows the recommened build configuration to get minimum footprint, but if you are and advanced user, you can configure as you prefer)"

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -0,0 +1 @@
+- Fix: microhttpd is now compiled by hand (version 0.9.48) and included in broker as a static lib (solves issue #1675)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: I'd suggest:

"Fix: libmicrohttpd 0.9.48 included in contextBroker as static lib (previous versions used 0.99.22 as dynamic library) (Issue #1675)"

(More compact and more informative)

File: rpm/SPECS/contextBroker.spec [hunk 3]
```diff
@@ -31,12 +31,12 @@ Release:   %{broker_release}
...
 Vendor:     Telefónica I+D
-Packager:   Fermín Galán <fermin@tid.es>
+Packager:   Fermín Galán <fermin.galanmarquez@telefonica.com>
 URL:        http://catalogue.fiware.org/enablers/publishsubscribe-context-broker-orion-context-broker
 Source:     %{name}-%{broker_version}.tar.gz
 BuildRoot: /var/tmp/%{name}-buildroot
-Requires:  libstdc++, boost-thread, boost-filesystem, libmicrohttpd, libcurl, logrotate
-Buildrequires: gcc, cmake, gcc-c++, libmicrohttpd-devel, libcurl-devel, boost-devel
+Requires:  libstdc++, boost-thread, boost-filesystem, gnutls, libcurl, logrotate
+Buildrequires: gcc, cmake, gcc-c++, gnutls-devel, libcurl-devel, boost-devel
```
Comment (fgalan) [inline rpm/SPECS/contextBroker.spec]: Removing libmicrohttpd as dependency has introduced gnutl, maybe due to a transient depedency previously hiden by libmicrohttpd. Maybe they are other dependencies hiden in that way... let's have a look to the dependencies of libmicrohttpd package itself to know.

Comment (fgalan) [inline rpm/SPECS/contextBroker.spec]: Some testing done at CentOS.

```
[fermin@centollo fiware-orion]$ rpm -q libmicrohttpd
libmicrohttpd-0.9.33-4.el6.x86_64
[fermin@centollo fiware-orion]$ rpm -q --requires libmicrohttpd
/sbin/ldconfig  
/sbin/ldconfig  
info  
info  
libc.so.6()(64bit)  
libc.so.6(GLIBC_2.10)(64bit)  
libc.so.6(GLIBC_2.2.5)(64bit)  
libc.so.6(GLIBC_2.3.2)(64bit)  
libc.so.6(GLIBC_2.3.4)(64bit)  
libc.so.6(GLIBC_2.4)(64bit)  
libc.so.6(GLIBC_2.9)(64bit)  
libdl.so.2()(64bit)  
libgcrypt.so.11()(64bit)  
libgcrypt.so.11(GCRYPT_1.2)(64bit)  
libgnutls.so.26()(64bit)  
libgnutls.so.26(GNUTLS_1_4)(64bit)  
libgpg-error.so.0()(64bit)  
libmicrohttpd.so.10()(64bit)  
libpthread.so.0()(64bit)  
libpthread.so.0(GLIBC_2.12)(64bit)  
libpthread.so.0(GLIBC_2.2.5)(64bit)  
librt.so.1()(64bit)  
librt.so.1(GLIBC_2.2.5)(64bit)  
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rtld(GNU_HASH)  
rpmlib(PayloadIsXz) <= 5.2-1
```

A little obscure... as rpm tool doesn't correlates this information to know to which RPM each file belongs.

Maybe an "empirical test" done in a just installed CentOS 6.x in FIWARE Lab once this PR gets merged would be more fruitful.

File: doc/manuals/admin/build_source.md (hunk 1 above)
Comment (fgalan) [inline doc/manuals/admin/build_source.md]: Fixed in 7a36466bef50bd8e4177e173432be63f912bdbdc

File: rpm/SPECS/contextBroker.spec (hunk 3 above)
Comment (fgalan) [inline rpm/SPECS/contextBroker.spec]: NTC in this PR (but test pending once it is merged)

File: CHANGES_NEXT_RELEASE (hunk 2 above)
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Fixed in d0c0b3bee0146d7b00f99fe4520243d1c80bc72d

**Comment by kzangeli** [comment]
LGTM

**Comment by crbrox** [comment]
LGTM

**Pull request merged**